clustered/client/src/main/java/org/ehcache/clustered/client/internal/loaderwriter/writebehind/ClusteredWriteBehindStore.java

```diff
@@ -0,0 +1,385 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.client.internal.loaderwriter.writebehind;
+
+import org.ehcache.clustered.client.internal.loaderwriter.ClusteredLoaderWriterStore;
+import org.ehcache.clustered.client.internal.store.ChainBuilder;
+import org.ehcache.clustered.client.internal.store.ClusteredStore;
+import org.ehcache.clustered.client.internal.store.ClusteredValueHolder;
+import org.ehcache.clustered.client.internal.store.ResolvedChain;
+import org.ehcache.clustered.client.internal.store.ServerStoreProxy;
+import org.ehcache.clustered.client.internal.store.lock.LockManager;
+import org.ehcache.clustered.client.internal.store.operations.ChainResolver;
+import org.ehcache.clustered.client.internal.store.operations.ConditionalRemoveOperation;
+import org.ehcache.clustered.client.internal.store.operations.ConditionalReplaceOperation;
+import org.ehcache.clustered.client.internal.store.operations.Operation;
+import org.ehcache.clustered.client.internal.store.operations.PutIfAbsentOperation;
+import org.ehcache.clustered.client.internal.store.operations.PutOperation;
+import org.ehcache.clustered.client.internal.store.operations.PutWithWriterOperation;
+import org.ehcache.clustered.client.internal.store.operations.RemoveOperation;
+import org.ehcache.clustered.client.internal.store.operations.ReplaceOperation;
+import org.ehcache.clustered.client.internal.store.operations.Result;
+import org.ehcache.clustered.client.internal.store.operations.codecs.OperationsCodec;
+import org.ehcache.clustered.client.service.ClusteringService;
+import org.ehcache.clustered.common.internal.store.Chain;
+import org.ehcache.clustered.common.internal.store.Element;
+import org.ehcache.config.ResourceType;
+import org.ehcache.core.spi.store.tiering.AuthoritativeTier;
+import org.ehcache.core.spi.time.TimeSource;
+import org.ehcache.core.spi.time.TimeSourceService;
+import org.ehcache.spi.loaderwriter.CacheLoaderWriter;
+import org.ehcache.spi.loaderwriter.WriteBehindConfiguration;
+import org.ehcache.spi.resilience.StoreAccessException;
+import org.ehcache.spi.service.ServiceConfiguration;
+import org.ehcache.spi.service.ServiceDependencies;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeoutException;
+
+import static org.ehcache.core.exceptions.StorePassThroughException.handleException;
+import static org.ehcache.core.spi.service.ServiceUtils.findSingletonAmongst;
+
+public class ClusteredWriteBehindStore<K, V> extends ClusteredStore<K, V> implements AuthoritativeTier<K, V> {
```

Comment:
I think we should be able to extend ClusteredLoaderWriterStore here and reuse a lot of code flow. wdyt?

Reply:
I don't see much code reuse here.

Reply:
AFAICT the `get` logic is same as that of clusteredloaderwriter.

Reply:
Got it, its different since we don't want to call `replaceAtHead`